tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,2211 @@
+         {'user': {'userName': 'admin1', 'userId': 4, 'roleId': 1},
+          'message': 'User has been created successfully'}),
+    ])
+    def test_create_user_with_password_token(self, foglamp_url, form_data, expected_values):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", "/foglamp/admin/user", body=json.dumps(form_data),
+                     headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert expected_values == jdoc
+
+    @pytest.mark.parametrize(("form_data", "expected_values"), [
+        ({"username": "any2", "password": "User@123"}, {'user': {'userName': 'any2', 'userId': 5, 'roleId': 2},
+                                                        'message': 'User has been created successfully'}),
+        ({"username": "admin2", "password": "F0gl@mp!", "role_id": 1},
+         {'user': {'userName': 'admin2', 'userId': 6, 'roleId': 1},
+          'message': 'User has been created successfully'}),
+    ])
+    def test_create_user_with_certificate_token(self, foglamp_url, form_data, expected_values):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", "/foglamp/admin/user", body=json.dumps(form_data),
+                     headers={"authorization": CERT_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert expected_values == jdoc
+
+    def test_update_password_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", "/foglamp/user/any1/password", body=json.dumps({"current_password": "User@123",
+                                                                            "new_password": "F0gl@mp1"}),
+                     headers={"authorization": PASSWORD_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'Password has been updated successfully for user id:<3>'} == jdoc
+
+    def test_update_password_with_certificate_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", "/foglamp/user/any2/password", body=json.dumps({"current_password": "User@123",
+                                                                            "new_password": "F0gl@mp1"}),
+                     headers={"authorization": CERT_TOKEN})
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert {'message': 'Password has been updated successfully for user id:<5>'} == jdoc
+
+    def test_reset_user_with_password_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
```

Comment:
add test snippet that user is able to login with newly reset password and not the old one